src/adapters/angle-protocol/products/savings/angleProtocolSavingsAdapter.ts

```diff
@@ -0,0 +1,387 @@
+      _input.protocolTokenAddress,
+      this.provider,
+    )
+
+    const filter = savingContract.filters.Withdraw(
+      undefined,
+      _input.userAddress,
+    )
+    const eventResults = await savingContract.queryFilter(
+      filter,
+      _input.fromBlock,
+      _input.toBlock,
+    )
+
+    return eventResults.map((event) => ({
+      blockNumber: event.blockNumber,
+      transactionHash: event.transactionHash,
+      protocolToken: token.protocolToken,
+      tokens: [
+        {
+          ...token.underlyingToken,
+          type: TokenType.Underlying,
+          balanceRaw: event.args?.assets,
+        },
+      ],
+    }))
+  }
+
+  async getDeposits(_input: GetEventsInput): Promise<MovementsByBlock[]> {
+    const tokens = await this.buildMetadata()
+    const token = tokens.find(
+      (t) => t.protocolToken.address === _input.protocolTokenAddress,
+    )!
+
+    const savingContract = Savings__factory.connect(
+      _input.protocolTokenAddress,
+      this.provider,
+    )
+
+    const filter = savingContract.filters.Deposit(undefined, _input.userAddress)
+    const eventResults = await savingContract.queryFilter(
+      filter,
+      _input.fromBlock,
+      _input.toBlock,
+    )
+
+    return eventResults.map((event) => ({
+      blockNumber: event.blockNumber,
+      transactionHash: event.transactionHash,
+      protocolToken: token.protocolToken,
+      tokens: [
+        {
+          ...token.underlyingToken,
+          type: TokenType.Underlying,
+          balanceRaw: event.args?.assets,
+        },
+      ],
+    }))
+  }
+
+  async getTotalValueLocked(
+    _input: GetTotalValueLockedInput,
+  ): Promise<ProtocolTokenTvl[]> {
+    const tokens = await this.buildMetadata()
+
+    return await Promise.all(
+      tokens.map(async (token) => {
+        const saving = Savings__factory.connect(
+          token.protocolToken.address,
+          this.provider,
+        )
+        const totalValueLocked = await saving.totalAssets({ blockTag: _input.blockNumber })
+        const totalSupply = await saving.totalSupply({ blockTag: _input.blockNumber })
+        return {
+          ...token.protocolToken,
+          type: TokenType.Protocol,
+          totalSupplyRaw: totalSupply,
+          tokens: [
+            {
+              ...token.underlyingToken,
+              type: TokenType.Underlying,
+              totalSupplyRaw: totalValueLocked,
+            },
+          ],
+        }
+      }),
+    )
+  }
+
+  async getProtocolTokenToUnderlyingTokenRate(
+    _input: GetConversionRateInput,
+  ): Promise<ProtocolTokenUnderlyingRate> {
+    const tokens = await this.buildMetadata()
+    const token = tokens.find(
+      (t) => t.protocolToken.address === _input.protocolTokenAddress,
+    )!
+
+    const savingsContract = Savings__factory.connect(
+      _input.protocolTokenAddress,
+      this.provider,
+    )
+    const underlyingRateRaw = await savingsContract.convertToAssets(
+      '1000000000000000000',
```

Comment:
We can't use etherjs with utils.parseEther()?

Reply:
what we need is the conversion from stEUR to agEUR so it won't be 1:1